src/main/java/org/spongepowered/common/mixin/core/world/MixinExplosion.java

```diff
@@ -410,8 +410,8 @@ public void doExplosionB(boolean spawnParticles) {
 
         if (this.causesFire) {
             for (BlockPos blockpos1 : this.affectedBlockPositions) {
-                if (this.world.getBlockState(blockpos1).getMaterial() == Material.AIR && this.world.getBlockState(blockpos1.down()).isFullBlock()
-                    && this.random.nextInt(3) == 0) {
+                if (this.random.nextInt(3) == 0 && this.world.getBlockState(blockpos1).getMaterial() == Material.AIR
```

Comment:
And here. you should be commenting why you're changing it (it's failing faster than checking the block state).

Reply:
Because there's no need to check the blockstate if `this.random.nextInt(3)` is not 0. I've moved the fastest condition first.